pkg/api/types.go

```diff
@@ -738,7 +771,35 @@ type Endpoints struct {
 	TypeMeta   `json:",inline"`
 	ObjectMeta `json:"metadata,omitempty"`
 
-	Endpoints []string `json:"endpoints,omitempty"`
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
+}
+
+type Endpoint struct {
+	// Required: The IP of this endpoint.
+	//FIXME: should we allow hostname-or-IP?  If so, maybe in other places.
+	IP string
+
+	// The ports exposed on this endpoint.
+	// FIXME: this would be nicer as a map of name -> struct
+	Ports []EndpointPort
+}
+
+type EndpointPort struct {
```

Comment:
If we change the name, we should make the same change to ContainerPort and ServicePort.  Given that the type name is not exposed to users, I'm OK to make the change, but I think it's churn for no value.